Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -67,6 +67,91 @@ struct TextureMapperLayer::ComputeTransformData {
     }
 };
 
+class TextureMapperFlattenedLayer final {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    explicit TextureMapperFlattenedLayer(const IntRect& rect, double zNear, double zFar)
+        : m_rect(rect)
+        , m_zNear(zNear)
+        , m_zFar(zFar)
+    { }
+
+    inline IntRect layerRect() const
+    {
+        return m_rect;
+    }
+
+    inline bool needsUpdate() const
+    {
+        return !m_textures.size();
+    }
+
+    void update(TextureMapperPaintOptions& options, const std::function<void(TextureMapperPaintOptions&)>& paintFunction)
+    {
+        double prevZNear, prevZFar;
+        options.textureMapper.getDepthRange(prevZNear, prevZFar);
+        options.textureMapper.setDepthRange(m_zNear, m_zFar);
+
+        m_textures.clear();
+
+        IntSize maxTextureSize = options.textureMapper.maxTextureSize();
+        tileLoop(maxTextureSize, [&](const IntRect& tileRect) {
+            RefPtr<BitmapTexture> surface = options.textureMapper.acquireTextureFromPool(tileRect.size(), { BitmapTexture::Flags::SupportsAlpha });
```

Comment:
auto

Reply:
This is on purpose. aquireTextureFromPool returns Ref<BitmapTexture> but we need RefPtr to make it work with SetForScope and TextureMapperPaintOptions which stores surface as RefPtr<BitmapTexture>. Otherwise we would need this Ref { *options.surface} line which was discussed before

Reply:
Ouch you've got me 😂 fair enough!